pkg/operator/controller/dns/controller.go

```diff
@@ -462,6 +476,11 @@ func (r *reconciler) createDNSProvider(dnsConfig *configv1.DNS, platformStatus *
 		return &dns.FakeProvider{}, nil
 	}
 
+	if infraStatus != nil && infraStatus.ControlPlaneTopology == configv1.ExternalTopologyMode {
```

Comment:
`infraStatus` cannot be nil.
```suggestion
	if infraStatus.ControlPlaneTopology == configv1.ExternalTopologyMode {
```